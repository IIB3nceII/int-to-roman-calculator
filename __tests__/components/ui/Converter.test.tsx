import { cleanup, render, screen } from "@testing-library/react";
import Home from "@/pages/index";
import { Converter } from "@/components/ui";

describe("Converter component tests", () => {
  beforeEach(() => {
    render(<Home />);
  });

  afterEach(() => {
    cleanup();
  });

  test("Output heading is visible on initialization.", () => {
    const outputHeading = screen.getByTestId("converter-output");

    expect(outputHeading).toBeInTheDocument();
  });

  test("Matches snapshot", () => {
    const { container } = render(<Converter />);
    expect(container).toMatchSnapshot();
  });
});
