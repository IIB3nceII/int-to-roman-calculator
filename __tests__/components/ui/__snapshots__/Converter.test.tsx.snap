// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converter component tests Matches snapshot 1`] = `
<div>
  <div>
    <input
      data-testid="converter-input"
      max="1000"
      min="1"
      placeholder="Write a number between 1 and 1000"
      type="number"
      value="1"
    />
    <h2
      data-testid="converter-output"
    >
      I
    </h2>
  </div>
</div>
`;
